src/main/java/com/ericsson/ei/subscription/RunSubscription.java

```diff
@@ -72,14 +72,14 @@ public boolean runSubscriptionOnObject(String aggregatedObject, Iterator<JsonNod
 
             if (id == null) {
                 LOGGER.error(
-                    "ID has not been passed for given aggregated object. The subscription will be triggered again.");
+                        "ID has not been passed for given aggregated object. The subscription will be triggered again.");
             }
 
-            if (subscriptionRepeatFlag == "false" && id != null
-                    && subscriptionRepeatDbHandler.checkIfAggrObjIdExistInSubscriptionAggrIdsMatchedList(
-                            subscriptionName, requirementIndex, id)) {
-                LOGGER.info("Subscription has already matched with AggregatedObject Id: {}\n"
-                        + "SubscriptionName: {}\nand has Subscription Repeat flag set to: {}",
+            if (subscriptionRepeatFlag == "false" && id != null && subscriptionRepeatDbHandler
```

Comment:
Booleans should not have == "false" use !subscriptionRepeatFlag. Second part of the statement should be moved out to it´s own boolean.

Reply:
It is not boolean but taken as string....

Anyway, It was like this before my pull request... for some reason converted to string and compared with "false"